x/protocolpool/keeper/msg_server.go

```diff
@@ -103,11 +105,101 @@ func (k MsgServer) CommunityPoolSpend(ctx context.Context, msg *types.MsgCommuni
 }
 
 func (k MsgServer) CreateContinuousFund(ctx context.Context, msg *types.MsgCreateContinuousFund) (*types.MsgCreateContinuousFundResponse, error) {
+	if err := k.validateAuthority(msg.Authority); err != nil {
+		return nil, err
+	}
+
+	recipient, err := k.Keeper.authKeeper.AddressCodec().StringToBytes(msg.Recipient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate the message fields
+	err = k.validateContinuousFund(ctx, *msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create continuous fund proposal
+	cf := types.ContinuousFund{
+		Recipient:  msg.Recipient,
+		Percentage: msg.Percentage,
+		Expiry:     msg.Expiry,
+	}
+
+	// Set continuous fund to the state
+	err = k.ContinuousFund.Set(ctx, recipient, cf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set recipient fund percentage & distribution
+	percentage := cf.Percentage.MulInt64(100)
+	err = k.RecipientFundPercentage.Set(ctx, recipient, percentage.TruncateInt())
+	if err != nil {
+		return nil, err
+	}
+	err = k.RecipientFundDistribution.Set(ctx, recipient, math.ZeroInt())
+	if err != nil {
+		return nil, err
+	}
+	err = k.ToDistribute.Set(ctx, math.ZeroInt())
```

Comment:
how come this is being set to 0? do we call iterate&distribute anywhere before this